script/tool/test/util.dart

```diff
@@ -11,32 +11,29 @@ import 'package:file/file.dart';
 import 'package:file/memory.dart';
 import 'package:flutter_plugin_tools/src/common.dart';
 import 'package:meta/meta.dart';
-import 'package:platform/platform.dart';
 import 'package:quiver/collection.dart';
 
-// TODO(stuartmorgan): Eliminate this in favor of setting up a clean filesystem
-// for each test, to eliminate the chance of files from one test interfering
-// with another test.
-FileSystem mockFileSystem = MemoryFileSystem(
-    style: const LocalPlatform().isWindows
-        ? FileSystemStyle.windows
-        : FileSystemStyle.posix);
-late Directory mockPackagesDir;
-
-/// Creates a mock packages directory in the mock file system.
+/// Creates a packages directory in the given location.
 ///
-/// If [parentDir] is set the mock packages dir will be creates as a child of
-/// it. If not [mockFileSystem] will be used instead.
-void initializeFakePackages({Directory? parentDir}) {
-  mockPackagesDir =
-      (parentDir ?? mockFileSystem.currentDirectory).childDirectory('packages');
-  mockPackagesDir.createSync();
+/// If [parentDir] is set the packages directory will be created there,
+/// otherwise [fileSystem] must be provided and it will be created an arbitrary
```

Comment:
I intentionally said that it would be created in "an arbitrary location" because tests shouldn't be written to assume anything about where in the in-memory filesystem will create the directory. They should save the return value if they need to reference it, not construct a (theoretically) matching path manually as some tests used to d.